lib/mail/utilities.rb

```diff
@@ -223,7 +223,26 @@ def map_with_index( enum, &block )
         enum.each_with_index.map(&block)
       end
 
-      def self.to_lf input
+    end
+    
+    # Test String#encode works correctly with line endings.
+    # Some versions of Ruby (e.g. MRI <1.9, JRuby, Rubinius) line ending
+    # normalization does not work correctly or did not have #encode.
+    if ("\r".encode(:universal_newline => true) rescue nil) == LF &&
+      (LF.encode(:crlf_newline => true) rescue nil) == CRLF
+      # Using String#encode is better performing than Regexp
+
+      def self.to_lf(input)
+        input.kind_of?(String) ? input.to_str.encode(:universal_newline => true) : ''
```

Comment:
This form of String#encoding will encode the string to whatever Encoding.default_internal is set to. When running a rails app, Encoding.default_internal is set to UTF-8

This new behaviour of changing the encoding is breaking some specs in my rails app, it was expecting the string to still be encoding as ASCII-8bit at this point.

We can fix this by doing `input.to_str.encode(input.encoding, :universal_newline => true)`

I can make a PR.